lib/absinthe/phase/document/result.ex

```diff
@@ -51,7 +51,19 @@ defmodule Absinthe.Phase.Document.Result do
     value =
       case Type.unwrap(emitter.schema_node.type) do
         %Type.Scalar{} = schema_node ->
-          Type.Scalar.serialize(schema_node, value)
+          try do
+            Type.Scalar.serialize(schema_node, value)
+          rescue
+            Absinthe.SerializationError ->
```

Comment:
I'm not in love with how raising the choice to have serializers succeed or raise puts us in this fun position of having to raise / rescue / raise the same exception but I don't really want to go down the rabbit hole of changing how all of this works right now, and what we're doing here to improve the error message doesn't really box us in for any future changes.